### PR TITLE
Address Regular Expression Denial of Service vulnerability

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Make changes to address the Catastrophic Backtracking security vulnerability by Synk. Regex processing time is linearly dependent on the length of the string, this attack makes use of arbitrary long strings to cause Denial of Service. 

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/channels/slack.py [hunk 1]
```diff
@@ -268,21 +268,23 @@ def _sanitize_user_message(
         uids_to_remove = uids_to_remove or []
 
         for uid_to_remove in uids_to_remove:
+            escaped_uid = re.escape(uid_to_remove)
```
Comment (radovanZRasa) [inline rasa/core/channels/slack.py:271]: We should add unit test which will check if re.escape is called.

File: changelog/12280.bugfix.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Addresses Regular Expression Denial of Service vulnerability in slack connector
```
Comment (ancalita) [inline changelog/12280.bugfix.md:1]: We should link to the CVE link if possible for more details.

File: rasa/core/channels/slack.py [hunk 3]
```diff
@@ -268,21 +268,23 @@ def _sanitize_user_message(
...
-                (rf"\s<@{uid_to_remove}>", ""),  # a bit arbitrary but probably OK
-                (rf"<@{uid_to_remove}>", " "),
+                (rf"<@{escaped_uid}>\s", ""),
+                (rf"\s<@{escaped_uid}>", ""),  # a bit arbitrary but probably OK
+                (rf"<@{escaped_uid}>", " "),
             ]:
                 text = re.sub(regex, replacement, text)
 
-        # Find multiple mailto or http links like
+        # Find multiple case-insensitive mailto or http links like
```
Comment (ancalita) [inline rasa/core/channels/slack.py:283]: Why did we need to change the pattern for finding case-insensitive links?

**Review by ancalita — COMMENTED** [review]
The failing slack unit test also must be investigated how it's connected to the made changes, I doubt it's flaky.

File: rasa/core/channels/slack.py (hunk 3 above)
Comment (vcidst) [reply rasa/core/channels/slack.py:283]: This change wasn't related to the security vulnerability. I just thought that a case-sensitive regex looks like a bug so I fixed it. Do you think I should isolate this change to a different ticket/PR for transparency?

Comment (vcidst) [reply rasa/core/channels/slack.py:283]: (the previous regex was case sensitive, it only works for URLs in lowercase)

Comment (ancalita) [reply rasa/core/channels/slack.py:283]: Yeah, I'd look in Jira if there has been a request for changing this before.
Also isolating would be a good idea to see which one of the two changes causes the unit test to fail.

File: changelog/12280.bugfix.md (hunk 2 above)
Comment (vcidst) [reply changelog/12280.bugfix.md]: The link to synk isn't public but I've added a link for RegEx Denial of Service

File: rasa/core/channels/slack.py (hunk 3 above)
Comment (vcidst) [reply rasa/core/channels/slack.py]: I've removed the case-insensitivity from the regex now

File: rasa/core/channels/slack.py (hunk 1 above)
Comment (vcidst) [reply rasa/core/channels/slack.py:270]: I've added a test to check for this

**Review by ancalita — APPROVED** [review]
👍🏻

**Pull request merged**